egenerator/manager/base.py

```diff
@@ -876,7 +868,7 @@ def train(
             # --------------------------
             # evaluate on validation set
             # --------------------------
-            if step % opt_config["validation_frequency"] == 0:
+            if step % opt_config["validation_frequency"] == 0 and step:  # not validate on the fist iteration
```

Comment:
I actually find this first point very interesting and important, since it provides the baseline of the untrained model

Reply:
I have, by the way, also tweaked the data input pipeline a bit in one of the branches feeding into the `CollectBreakingChanges`. Some of the work is now distributed further to the worker nodes, such that it is more easily parallelized, which will speed up the data pipeline. If the reason for you to make this change was to reduce the setup time until training starts, you could use this newer branch and reduce the number of additional files that you require per event pool in the validation data iterator or increase the number of workers to match the number of files. That way you only need one iteration of file loading until the queue is populated

Reply:
Great! Okay I will revert this than.